packages/cli/src/config/extension.test.ts

```diff
@@ -1111,6 +1111,124 @@ This extension will run the following MCP servers:
       );
     });
 
+    it('should only prompt for new settings on update', async () => {
+      // 1. Create and install the "old" version of the extension.
+      const oldSourceExtDir = createExtension({
+        extensionsDir: tempHomeDir, // Create it in a temp location first
+        name: 'my-local-extension',
+        version: '1.0.0',
+        settings: [
+          {
+            name: 'API Key',
+            description: 'Your API key for the service.',
+            envVar: 'MY_API_KEY',
+          },
+        ],
+      });
+
+      // Install it so it exists in the userExtensionsDir
+      await installOrUpdateExtension(
+        { source: oldSourceExtDir, type: 'local' },
+        async (_) => true,
+        process.cwd(),
+      );
+
+      // 2. Create the "new" version of the extension in a new source directory.
+      const newSourceExtDir = createExtension({
+        extensionsDir: tempHomeDir, // Another temp location
+        name: 'my-local-extension', // Same name
+        version: '1.1.0', // New version
+        settings: [
+          {
+            name: 'API Key',
+            description: 'Your API key for the service.',
+            envVar: 'MY_API_KEY',
+          },
+          {
+            name: 'New Setting',
+            description: 'A new setting.',
+            envVar: 'NEW_SETTING',
+          },
+        ],
+      });
+
+      const previousExtensionConfig = loadExtensionConfig({
+        extensionDir: path.join(userExtensionsDir, 'my-local-extension'),
+        workspaceDir: process.cwd(),
+        extensionEnablementManager: new ExtensionEnablementManager(),
+      });
+
+      const promptForSettingsMock = vi.fn(
+        async (_: ExtensionSetting): Promise<string> => Promise.resolve(''),
+      );
+
+      // 3. Call installOrUpdateExtension to perform the update.
+      await installOrUpdateExtension(
+        { source: newSourceExtDir, type: 'local' },
+        async (_) => true,
+        process.cwd(),
+        previousExtensionConfig,
+        promptForSettingsMock,
+      );
+
+      expect(promptForSettingsMock).toHaveBeenCalledTimes(1);
+      expect(promptForSettingsMock).toHaveBeenCalledWith(
+        expect.objectContaining({ name: 'New Setting' }),
+      );
```

Comment:
This should also validate that both settings are now set (and we didn't lose the previous one)